runtime/runtime/src/lib.rs

```diff
@@ -1070,6 +1072,48 @@ impl Runtime {
         Ok(())
     }
 
+    // In test runs reads and writes here used 442 TGas. Added 10% to account for possible bigger
+    // state
+    #[cfg(feature = "protocol_feature_fix_storage_usage")]
+    const GAS_USED_FOR_STORAGE_USAGE_DELTA_MIGRATION: Gas = 490_000_000_000_000;
+
+    pub fn apply_migrations(
+        &self,
+        state_update: &mut TrieUpdate,
+        apply_state: &ApplyState,
+    ) -> Result<Gas, StorageError> {
+        #[cfg(feature = "protocol_feature_fix_storage_usage")]
+        let mut gas_used: Gas = 0;
+        #[cfg(not(feature = "protocol_feature_fix_storage_usage"))]
+        let gas_used: Gas = 0;
+        #[cfg(feature = "protocol_feature_fix_storage_usage")]
+        if ProtocolFeature::FixStorageUsage.protocol_version()
+            == apply_state.current_protocol_version
+        {
+            for (account_id, delta) in
+                &apply_state.migration_context.migration_data.storage_usage_delta
+            {
+                match get_account(state_update, account_id)? {
+                    Some(mut account) => {
+                        // Storage usage is saved in state, hence it is nowhere close to max value
+                        // of u64, and maximal delta is 4196, se we can add here without checking
+                        // for overflow
+                        account.set_storage_usage(account.storage_usage() + delta);
+                        set_account(state_update, account_id.clone(), &account);
+                    }
+                    // Account could have been deleted in the meantime
+                    None => {}
+                }
+            }
+            gas_used += Runtime::GAS_USED_FOR_STORAGE_USAGE_DELTA_MIGRATION;
```

Comment:
Do we have to hardcode this value instead of computing it as we apply the migrations?

Reply:
I am not really sure how. For my calculations I did everything through VMLogic